internal/controller/progressive/progressive.go

```diff
@@ -79,6 +86,15 @@ func ProcessResource(
 			return false, false, 0, fmt.Errorf("error getting %s: %v", currentUpgradingChildDef.GetKind(), err)
 		}
 		if currentUpgradingChildDef == nil {
+			// Scale down the promoted child first
+			scaledDown, err := scaleDownPromotedChild(ctx, rolloutObject, liveRolloutObject, existingPromotedChild, skipPromotedChildSourceVerticesScaling, controller, c)
+			if err != nil {
+				return false, false, 0, err
+			}
+			if scaledDown {
+				return false, false, common.DefaultRequeueDelay, nil
+			}
+
```

Comment:
This can be an optimization for later, but I'm thinking it may be better to successfully create the new one before we scale down.